### PR TITLE
website/integrations: gitlab: add oidc guide

### LINKED ISSUE #8175: GitLab SAML vs OIDC Docs
**Describe your question/**

According to https://goauthentik.io/integrations/services/gitlab/ one should configure GitLab with a SAML provider. This comes with some caveats for me. E.g.

* The cert fingerprint does expire with the cert itself and i need to update it regulary (This may be a problem with my ignorance towards SAML)
* IMHO: Its less intuitive to configure compared to OIDC

I just tried to achieve SSO via OIDC by following https://docs.gitlab.com/ee/administration/auth/oidc.html
It just worked fine and there were no big hurdles.

example config:

```
        gitlab_rails['omniauth_allow_single_sign_on'] = ['openid_connect']
        gitlab_rails['omniauth_sync_email_from_provider'] = 'openid_connect'
        gitlab_rails['omniauth_sync_profile_from_provider'] = ['openid_connect']
        gitlab_rails['omniauth_sync_profile_attributes'] = ['email']
        gitlab_rails['omniauth_auto_sign_in_with_provider'] = 'openid_connect'
        gitlab_rails['omniauth_block_auto_created_users'] = false
        gitlab_rails['omniauth_auto_link_saml_user'] = true
        gitlab_rails['omniauth_auto_link_user'] = ["openid_connect"]
        gitlab_rails['omniauth_providers'] = [
          {
            name: 'openid_connect',
            label: 'My Company OIDC Login',
            args: {
              name: 'openid_connect',
              scope: ['openid','profile','email'],
              response_type: 'code',
              issuer: 'https://authentik.mycompany/application/o/gitlab-slug/',
              discovery: true,
              client_auth_method: 'query',
              uid_field: 'preferred_username',
              send_scope_to_token_endpoint: 'true',
              pkce: true,
              client_options: {
                identifier: '${OIDC_CLIENT_ID}',
                secret: '${OIDC_CLIENT_SECRET}',
                redirect_uri: 'https://gitlab.mycompany/users/auth/openid_connect/callback'
              }
            }
          }
        ]
```

Is there any reason why the Docs prefere/suggest saml?
If no, can i help in any way to provide a OIDC how-to for gitlab?

### PR DESCRIPTION
**Pull request by motey**
## Details

Adding a guide to connect GitLab via OIDC (alternative to SAML which is the only guide at the moment)

Discussed in:

Closes #8175

---

## Checklist

-   [ ] Local tests pass (`ak test authentik/`) <- not applicable; only doc changes
-   [ ] The code has been formatted (`make lint-fix`) <- not applicable; only doc changes

If an API change has been made

-   [ ] The API schema has been updated (`make gen-build`) <- not applicable; only doc changes

If changes to the frontend have been made

-   [ ] The code has been formatted (`make web`) <- not applicable; only doc changes

If applicable

-   [x] The documentation has been updated
-   [x] The documentation has been formatted (`make website`) <- Yes, but very unsure about this. should that not happen on "server"-side (github actions) in a controlled environment. Needed to install a lot of stuff on my machine for that :D not very contributor friendly

### CODE REVIEW

File: website/integrations/services/gitlab/index.md [hunk 1]
```diff
@@ -62,3 +75,61 @@ gitlab_rails['omniauth_providers'] = [
...
+-   Redirect URI/Origins: `https://gitlab.company/users/auth/openid_connect/callback`
+-   Scopes: `email`, `openid`, `profile`
+-   Subject mode: `Based on the Users's Email`
+-   Include claims in id_token: `True`
+
+Under _Advanced protocol settings_, set a certificate for _Signing Certificate_.
+
+#### GitLab Configuration
+
+Paste the following block in your `gitlab.rb` file, after replacing the placeholder values from above. The file is located in `/etc/gitlab`.
```
[the remark this answers is not in the document]
Comment (motey) [reply website/integrations/services/gitlab/index.md]: Good point, i just copied the sentence from the older SAML guide. Fixed it in the SAML guide as well with https://github.com/goauthentik/authentik/pull/9740/commits/de2d1df4430d629601e08cc6faf88ce71aca8958

**Review by rissson — APPROVED** [review]

**Review by dominic-r — CHANGES_REQUESTED** [review]

File: website/integrations/services/gitlab/index.md [hunk 2]
```diff
@@ -62,3 +75,61 @@ gitlab_rails['omniauth_providers'] = [
 ```
 
 Afterwards, either run `gitlab-ctl reconfigure` if you're running GitLab Omnibus, or restart the container if you're using the container.
+
+### OpenID Connect auth
+
+#### Preparation
+
+The following placeholders will be used:
```
[the remark this answers is not in the document]
Comment (motey) [reply website/integrations/services/gitlab/index.md:83]: But the scope is the current chapter. And there is nothing defined in this scope. 
In the real world people will jump to this chapter and not read everything first.

[the remark this answers is not in the document]
Comment (motey) [reply website/integrations/services/gitlab/index.md:83]: >  Placeholders are not defined twice on other pages

Aint true in my world;
see https://docs.goauthentik.io/integrations/services/nextcloud/#openid-connect-auth vs https://docs.goauthentik.io/integrations/services/nextcloud/#saml-auth
If i am remembering correct, this is also what i took as template. And i just took the time to browse through all integrations docs. For now Nextcloud is the only one with 2 integration options. So i am not sure on what your statement is based. Could you provide an example?

> To be honest that's not really our problems if users are unable to read things

Ok, fair enough that is an opinion. But what is the advantage of having it only on the top and what is the disadvantage of defining it twice to have chapters to be self-consistent?

[the remark this answers is not in the document]
Comment (tanberry) [reply website/integrations/services/gitlab/index.md:83]: Yeah, let's leave the placeholders and the Placeholder section for now...

**Review by tanberry — CHANGES_REQUESTED** [review]
Thanks so much for this PR! And apologies; looks like I starting reviewing it weeks ago but never submitted it, so it has been sitting in Pending mode for a while, sorry.

**Comment by motey** [comment]
@tanberry sorry, i am a little bit confused, whats the state of this PR is now and whose turn it is now :) maybe you can help out. thanks.

**Review by tanberry — APPROVED** [review]
Thanks again @motey and it's me that needs to apologize for not getting back to this earlier! Let's merge!

**Pull request merged**